lnbits/extensions/boltz/crud.py

```diff
@@ -172,54 +172,162 @@ async def create_reverse_submarine_swap(
         VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
         """,
         (
-            swap.id,
+            reverse_swap.id,
+            reverse_swap.wallet,
+            reverse_swap.status,
+            reverse_swap.boltz_id,
+            reverse_swap.instant_settlement,
+            reverse_swap.preimage,
+            reverse_swap.claim_privkey,
+            reverse_swap.lockup_address,
+            reverse_swap.invoice,
+            reverse_swap.onchain_amount,
+            reverse_swap.onchain_address,
+            reverse_swap.timeout_block_height,
+            reverse_swap.redeem_script,
+            reverse_swap.amount,
+        ),
+    )
+
+    claim_task = asyncio.create_task(
+        client.claim_reverse_swap(
+            privkey_wif=claim_privkey_wif,
+            preimage_hex=preimage_hex,
+            lockup_address=swap.lockupAddress,
+            receive_address=data.onchain_address,
+            redeem_script_hex=swap.redeemScript,
+        )
+    )
+
+    pay_task = pay_invoice_and_update_status(
+        swap_id,
+        claim_task,
+        pay_invoice(
+            wallet_id=data.wallet,
+            payment_request=swap.invoice,
+            description=f"reverse swap for {swap.onchainAmount} sats on boltz.exchange",
+            extra={"tag": "boltz", "swap_id": swap_id, "reverse": True},
+        ),
+    )
+
+    asyncio.gather(claim_task, pay_task)
+
+    return reverse_swap
+
+
+def pay_invoice_and_update_status(
+    swap_id: str, wstask: asyncio.Task, awaitable: Awaitable
+) -> asyncio.Task:
+    async def _pay_invoice(awaitable):
+        try:
+            awaited = await awaitable
+            await update_swap_status(swap_id, "complete")
+            return awaited
+        except asyncio.exceptions.CancelledError:
+            """lnbits process was exited, do nothing and handle it in startup script"""
+        except:
+            wstask.cancel()
+            await update_swap_status(swap_id, "failed")
+
+    return asyncio.create_task(_pay_invoice(awaitable))
+
+
+async def get_auto_reverse_submarine_swaps(
+    wallet_ids: Union[str, List[str]]
+) -> List[AutoReverseSubmarineSwap]:
+    if isinstance(wallet_ids, str):
+        wallet_ids = [wallet_ids]
```

Comment:
I would use `list[str]` only and call the function properly instead of treating the two cases.

Reply:
2c0cd5a8